### PR TITLE
Update benchmarks and domainslib to support ocaml 4.14.0+domains (ocaml 5.0)

### PR DESCRIPTION
**Pull request by shubhamkumar13**
This PR is the first step in adding ocaml 5.0 support for the benchmarks in sandmark.

The PR adds domainslib.0.3.2 and updates the benchmarks supporting the updated api for domainslib

### CODE REVIEW

**Comment by shakthimaan** [comment]
```
# OCaml version mismatch: 5.00.0, expected 4.14.0
```
The version number mismatch is causing the 4.14.0+trunk build to fail. The following issue is related: https://github.com/ocaml-multicore/ocaml-multicore/pull/709.

**Comment by kayceesrk** [comment]
https://github.com/ocaml-multicore/ocaml-multicore/pull/709 is not relevant as it has been closed. There should be no mention of 5.00 in any of these since 5.00 has not been tagged. 

Where does the string "5.00" come from? That should be changed to 4.14(+...).

**Comment by kayceesrk** [comment]
Looks like upstream has changed the version number to 5.00.0: https://github.com/ocaml/ocaml/blob/trunk/VERSION#L1. So `4.14.0+trunk` should be `5.00.0+trunk`.

**Comment by kayceesrk** [comment]
Ping @shubhamkumar13.

**Comment by shubhamkumar13** [comment]
I've added an issue tracing `irmin` benchmarks failing on `ocaml.5.00.0+trunk` which would be fixed when the trunk is frozen (which would be easier for generating ppxlib version supporting trunk).

I think it's best to merge this before and then https://github.com/ocaml-bench/sandmark/pull/257 can be merged (since there are changes to drome.yml file where I've updated the trunk from `4.14.0+trunk` to `5.00.0+trunk`)

**Comment by shubhamkumar13** [comment]
This PR also updates `drone.yml` file to 5.00. 
This is in addition to the adding the updated domainslib and making benchmarks compatible to 5.00

File: Makefile [hunk 1]
```diff
@@ -113,6 +113,7 @@ ocaml-versions/%.bench: check_url depend log_sandmark_hash ocaml-versions/%.json
 		*multicore*) opam install --switch=$* --keep-build-dir --yes ocaml-migrate-parsetree.2.1.0+multicore ppxlib.0.22.0+multicore ;; \
 		*effects*) opam install --switch=$* --keep-build-dir --yes ocaml-migrate-parsetree.2.1.0+multicore ppxlib.0.22.0+multicore ;; \
 		*domains*) opam install --switch=$* --keep-build-dir --yes ocaml-migrate-parsetree.2.2.0+stock ppxlib.0.22.0+stock ;; \
+		*5.00*) sed 's/(alias (name buildbench) (deps layers.exe irmin_mem_rw.exe))/; (alias (name buildbench) (deps layers.exe irmin_mem_rw.exe))/g' ./benchmarks/irmin/dune > ./benchmarks/irmin/dune ;; \
```
Comment (shakthimaan) [inline Makefile:116]: The build still tries to install `ocaml-migrate-parse-tree.2.2.0+stock`. Also, in the `clean` target in the Makefile, you need to undo this change if it was modified.

Comment (shubhamkumar13) [reply Makefile]: Thanks @shakthimaan for the feedback!
Updated the clean target to reset the changes `benchmarks/irmin/dune` and added the deps in the switch case to ensure any package which depends on ppxlib (which depends on omp) to ensure that `ocaml-migrate-parsetree.2.2.0+stock` doesn't attempt an install

File: .drone.yml [hunk 2]
```diff
@@ -18,32 +18,7 @@ steps:
...
-  - pip3 install intervaltree
-  - sudo chown -R opam .
-  - eval $(opam env)
-  - opam update
-  - opam install dune.2.9.0
-  - export ITER=1
-  - export OPAM_DISABLE_SANDBOXING=true
-  - TAG='"run_in_ci"' make run_config_filtered.json
-  - RUN_CONFIG_JSON=run_config_filtered.json make ocaml-versions/4.12.0+stock.bench
+  - RUN_CONFIG_JSON=run_config_filtered.json make ocaml-versions/5.00.0+trunk.bench
```
[the remark this answers is not in the document]
Comment (shubhamkumar13) [reply .drone.yml:21]: Just did! Thanks!

[the remark this answers is not in the document]
Comment (shakthimaan) [reply .drone.yml:21]: Please test the parallel entry and add the same to .drone.yml and this PR is ready to be merged. Thanks!
